src/mock.ts

```diff
@@ -91,57 +101,68 @@ function addMockFunctionsToSchema({ schema, mocks = {}, preserveResolvers = fals
   // and completes the customMock object with any fields
   // defined on genericMock
   // only merges objects or arrays. Scalars are returned as is
-  function mergeMocks(genericMockFunction, customMock) {
+  function mergeMocks(genericMockFunction: IMockFn, customMock: any): any {
     if (Array.isArray(customMock)) {
-      return customMock.map((el) => mergeMocks(genericMockFunction, el));
+      return customMock.map((el: any) => mergeMocks(genericMockFunction, el));
     }
     if (isObject(customMock)) {
       return mergeObjects(genericMockFunction(), customMock);
     }
     return customMock;
   }
 
-  function assignResolveType(type) {
+  function getResolveType(namedFieldType: GraphQLNamedType) {
+    if ( (namedFieldType instanceof GraphQLInterfaceType) ||
+         (namedFieldType instanceof GraphQLUnionType) ) {
+        return namedFieldType.resolveType;
+    } else {
+        return undefined;
+    }
+  }
+
+  function assignResolveType(type: GraphQLType) {
     const fieldType = getNullableType(type);
     const namedFieldType = getNamedType(fieldType);
 
-    const oldResolveType = namedFieldType.resolveType;
+    const oldResolveType = getResolveType(namedFieldType);
```

Comment:
I think this may no longer work with graphql 0.7.0, which adds a "fake" resolveType to unions and interfaces generated with buildSchema.